spec/unit/util/execution_spec.rb

```diff
@@ -219,6 +244,36 @@ def stub_process_wait(exitstatus)
           end
         end
 
+        describe "cwd option" do
+          def expect_cwd_to_be(cwd)
+            Puppet::Util::Execution.expects(executor).with(
+              anything,
+              has_entries(:cwd => cwd),
+              anything,
+              anything,
+              anything
+            ).returns(rval)
+          end
+
+          it 'should raise an ArgumentError if the specified working directory does not exist' do
+            cwd = 'cwd'
+            Puppet::FileSystem.expects(:directory?).with(cwd).returns(false)
+
+            expect {
+              Puppet::Util::Execution.execute('test command', cwd: cwd)
+            }.to raise_error do |error|
+              expect(error).to be_a(ArgumentError)
+              expect(error.message).to match(cwd)
+            end
+          end
+
+          it "should set the cwd to the user-specified one" do
+            Puppet::FileSystem.stubs(:directory?).with('cwd').returns(true)
```

Comment:
no need for the `.with('cwd')`

Reply:
The `with` ensures we return `true` only for our cwd.

Reply:
Ah